pkg/boot/universalpayload/utilities_arch_arm64.go

```diff
@@ -25,48 +24,86 @@ func getPhysicalAddressSizes() (uint8, error) {
 // bootloader parameter needs to be prepared in trampoline code.
 // Also stack is prepared in trampoline code snippet to ensure no data leak.
 func constructTrampoline(buf []uint8, hobAddr uint64, entry uint64) []uint8 {
-	ptrToSlice := func(ptr uintptr, size int) []byte {
-		var data []byte
-
-		sh := (*reflect.SliceHeader)(unsafe.Pointer(&data))
-		sh.Data = ptr
-		sh.Len = size
-		sh.Cap = size
-
-		return data
+	// In order to support Position Indepent Execution, we need to update actual
+	// physical address of HoB list, stack, and entry point of UPL FIT image.
+	//
+	// Due to Plan 9 Assembly limitation, we can only get symbol address instead
```

Comment:
I've written plan 9 assembly code for a  long time now, and I am pretty sure what you want to do is possible. But maybe there are additional go assembly limitations. It would be nice to look at this problem someday and make sure it is really a problem.

Reply:
Thanks for your comments @rminnich :),  we found this issue when using different Golang toolchain. I will deep dive into source code of plan 9 assembly support on Golang to figure out general solution for this issue in next pull request.